java/spring/containerWithTests/Application/sampleWebApp.FunctionalTests/src/main/java/functional_tests/SampleFunctionalTest.java

```diff
@@ -43,12 +47,11 @@ public void testAssertTitleWithSelenium() throws InterruptedException, Assertion
             }
             catch(AssertionError e)
             {
-                if(i == (numRetries - 1))
-                {
+                if(currentTimestamp > endTimestamp) {

```

Comment:
why are you using the old value of currentTimestamp here

Reply:
So exception should be thrown if test runs for more than 10 minutes, endTimestamp is calculated as initial currentTimestamp + 10 minutes. With 10 seconds as buffer, if currentTimestamp is > endTimestamp, we throw error